sim/druid/talents.go

```diff
@@ -45,25 +57,60 @@ func (druid *Druid) ApplyTalents() {
 		if druid.InForm(Cat) {
 			druid.MultiplyStat(stats.AttackPower, 1.0+0.5*bonus)
 		} else if druid.InForm(Bear) {
-			druid.MultiplyStat(stats.Stamina, 1.0+bonus)
+			druid.MultiplyStat(stats.Stamina, 1.0+0.5*bonus)
 		}
 	}
 
 	if druid.Talents.SurvivalOfTheFittest > 0 {
-		bonus := 0.01 * float64(druid.Talents.SurvivalOfTheFittest)
+		bonus := 0.02 * float64(druid.Talents.SurvivalOfTheFittest)
+		druid.MultiplyStat(stats.Stamina, 1.0+bonus)
+		druid.MultiplyStat(stats.Strength, 1.0+bonus)
+		druid.MultiplyStat(stats.Agility, 1.0+bonus)
+		druid.MultiplyStat(stats.Intellect, 1.0+bonus)
+		druid.MultiplyStat(stats.Spirit, 1.0+bonus)
+		druid.PseudoStats.ReducedCritTakenChance += 0.02 * float64(druid.Talents.SurvivalOfTheFittest)
+		if druid.InForm(Bear) {
+			druid.AddStat(stats.Armor, druid.Equip.Stats()[stats.Armor]*(0.33/3)*float64(druid.Talents.ThickHide))
+		}
+	}
+
+	if druid.Talents.ImprovedMarkOfTheWild > 0 {
+		bonus := 0.01 * float64(druid.Talents.ImprovedMarkOfTheWild)
 		druid.MultiplyStat(stats.Stamina, 1.0+bonus)
 		druid.MultiplyStat(stats.Strength, 1.0+bonus)
 		druid.MultiplyStat(stats.Agility, 1.0+bonus)
 		druid.MultiplyStat(stats.Intellect, 1.0+bonus)
 		druid.MultiplyStat(stats.Spirit, 1.0+bonus)
-		druid.PseudoStats.ReducedCritTakenChance += 0.01 * float64(druid.Talents.SurvivalOfTheFittest)
+	}
+
+	if druid.Talents.ProtectorOfThePack > 0 {
+		bonus := 0.02 * float64(druid.Talents.ProtectorOfThePack)
+		if druid.InForm(Bear) {
+			druid.MultiplyStat(stats.AttackPower, 1.0+bonus)
+			druid.PseudoStats.DamageTakenMultiplier -= 0.04 * float64(druid.Talents.ProtectorOfThePack)
+		}
+	}
+
+	if druid.Talents.PrimalPrecision > 0 {
+		druid.AddStat(stats.Expertise, 5.0*float64(druid.Talents.PrimalPrecision))
 	}
 
 	if druid.Talents.LivingSpirit > 0 {
 		bonus := 0.05 * float64(druid.Talents.LivingSpirit)
 		druid.MultiplyStat(stats.Spirit, 1.0+bonus)
 	}
 
+	if druid.Talents.MasterShapeshifter > 0 {
+		bonus := 0.02 * float64(druid.Talents.MasterShapeshifter)
+		if druid.InForm(Moonkin) {
+			druid.MultiplyStat(stats.SpellPower, 1.0+bonus)
+		} else if druid.InForm(Bear) {
+			druid.PseudoStats.DamageDealtMultiplier += bonus
+		} else if druid.InForm(Cat) {
+			druid.PseudoStats.BonusMeleeCritRating += 2 * float64(druid.Talents.MasterShapeshifter) * core.CritRatingPerCritChance
```

Comment:
This can just be `druid.AddStat(stats.MeleeCrit, ....)`. That pseudostat is specifically for non-ranged crit.

Reply:
shoot yea, somehow it didnt occur to me thered be a `stats.MeleeCrit`